modules/etc/image_etc.cpp

```diff
@@ -47,13 +47,14 @@ static Image::Format _get_etc2_mode(Image::DetectChannels format) {
 		case Image::DETECTED_RGB:
 			return Image::FORMAT_ETC2_RGB8;
 
-		default:
+		case Image::DETECTED_RGBA:
 			return Image::FORMAT_ETC2_RGBA8;
 
-			// TODO: would be nice if we could use FORMAT_ETC2_RGB8A1 for FORMAT_RGBA5551
+		// TODO: would be nice if we could use FORMAT_ETC2_RGB8A1 for FORMAT_RGBA5551
+		default:
+			// TODO: Kept for compatibility, but should be investigated whether it's correct or if it should error out
+			return Image::FORMAT_ETC2_RGBA8;
```

Comment:
@tagcup @reduz The error condition would never be reached, so I removed it and kept the "everything else is RGBA8" default. I don't know if it's correct but I kept it for compatibility.

I only came to this file due to warnings in the next method:
```
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_BPTC_RGBA' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_BPTC_RGBF' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_BPTC_RGBFU' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_DXT1' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_DXT3' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_DXT5' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_L8' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_LA8' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_MAX' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_PVRTC2A' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_PVRTC2' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_PVRTC4A' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_PVRTC4' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_R8' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_RF' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_RG8' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_RGB8' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_RGBA4444' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_RGBA5551' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_RGBA8' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_RGBAF' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_RGBAH' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_RGBE9995' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_RGBF' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_RGBH' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_RGF' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_RGH' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_RGTC_RG' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_RGTC_R' not handled in switch [-Wswitch]
modules/etc/image_etc.cpp:60:9: warning: enumeration value 'FORMAT_RH' not handled in switch [-Wswitch]
```